### PR TITLE
Nested source mapping

### LINKED ISSUE #53: Mapping of nested objects
**Is your feature request related to a problem? Please describe.**

Treeview (and probably some other future elements) require mapping of nested source data.

**Describe the solution you'd like**

Implemented recursive loop and mapping through nested items

### PR DESCRIPTION
**Pull request by Sneki**
Implemented nested source mapping (necessary for treeview data)

resolves #53

### CODE REVIEW

File: src/utility/mapping.js [hunk 1]
```diff
@@ -1,33 +1,26 @@
...
-There is a lot of cloning here to handle mapping.
-When implementing nesting we need to use cloneDeep.
-This method should be optimized if possible as it
-would be slow on large datasets. Test performance.
-*/
 const mapItem = (schema, item) => {
   /* eslint no-param-reassign:"off" */
   each(schema, (field) => {
-    item[`${field.name}${originalSuffix}`] = clone(item[field.name]);
+    item[`${field.name}${originalSuffix}`] = item[field.name];
```
Comment (manico) [inline src/utility/mapping.js]: Clone was added because of mutations so that original is left unchanged. Will this be handled by this change?

Comment (Sneki) [reply src/utility/mapping.js]: I can add it. It works fine with it, just couldn't find the reason why was added in the first place..

Comment (Lanchi) [reply src/utility/mapping.js]: From what I can remember, we had original and mapped properties duplicated, so original are being deleted but references should be broken before deleting props.

https://github.com/chmjs/chameleon-vuetify/issues/45
https://github.com/chmjs/chameleon-sdk/pull/24

Comment (Sneki) [reply src/utility/mapping.js]: Yes, that's the thing I tested and it seems fine now even without clone. What am i missing 🤔

**Comment by Lanchi** [comment]
There's even an old bug on mapping, not related to usage of `clone`. 
If mapping is set like this
screenshot 2018-12-18 at 15 32 25
SDK returns data like this
screenshot 2018-12-18 at 15 32 52

**Pull request merged**